src/integration-tests/test_admin_client.py

```diff
@@ -108,6 +156,234 @@ def test_breathing(single_node: Cluster) -> None:
+    producer: Client = proxy.create_client("producer")
+
+    task = PostRecord(tc.DOMAIN_FANOUT, "test_stats", num=32)
+    post_n_msgs(producer, task)
+
+    stats = extract_stats(admin.send_admin("encoding json_pretty stat show"))
+    queue_stats = stats["domainQueues"]["domains"][tc.DOMAIN_FANOUT][task.uri]
+
+    expected_stats_after_post = {
+        "appIds": {
+            "bar": {
+                "values": {
+                    "queue_confirm_time_max": 0,
+                    "queue_confirm_time_avg": 0,
+                    "queue_nack_msgs": 0,
+                    "queue_ack_time_max": 0,
+                    "queue_ack_msgs": 0,
+                    "queue_confirm_msgs": 0,
+                    "queue_push_bytes": 0,
+                    "queue_consumers_count": 0,
+                    "queue_producers_count": 0,
+                    "queue_push_msgs": 0,
+                    "queue_ack_time_avg": 0,
+                    "queue_put_bytes": 0,
+                    "queue_put_msgs": 0,
+                }
+            },
+            "baz": {
+                "values": {
+                    "queue_confirm_time_max": 0,
+                    "queue_confirm_time_avg": 0,
+                    "queue_nack_msgs": 0,
+                    "queue_ack_time_max": 0,
+                    "queue_ack_msgs": 0,
+                    "queue_confirm_msgs": 0,
+                    "queue_push_bytes": 0,
+                    "queue_consumers_count": 0,
+                    "queue_producers_count": 0,
+                    "queue_push_msgs": 0,
+                    "queue_ack_time_avg": 0,
+                    "queue_put_bytes": 0,
+                    "queue_put_msgs": 0,
+                }
+            },
+            "foo": {
+                "values": {
+                    "queue_confirm_time_max": 0,
+                    "queue_confirm_time_avg": 0,
+                    "queue_nack_msgs": 0,
+                    "queue_ack_time_max": 0,
+                    "queue_ack_msgs": 0,
+                    "queue_confirm_msgs": 0,
+                    "queue_push_bytes": 0,
+                    "queue_consumers_count": 0,
+                    "queue_producers_count": 0,
+                    "queue_push_msgs": 0,
+                    "queue_ack_time_avg": 0,
+                    "queue_put_bytes": 0,
+                    "queue_put_msgs": 0,
+                }
+            },
+        },
+        "values": {
+            "queue_confirm_time_max": 0,
+            "queue_confirm_time_avg": 0,
+            "queue_nack_msgs": 0,
+            "queue_ack_time_max": GreaterThan(0),
+            "queue_ack_msgs": 32,
+            "queue_confirm_msgs": 0,
+            "queue_push_bytes": 0,
+            "queue_consumers_count": 0,
+            "queue_producers_count": 0,
+            "queue_push_msgs": 0,
+            "queue_ack_time_avg": GreaterThan(0),
+            "queue_put_bytes": 96,
+            "queue_put_msgs": 32,
+        },
+    }
```

Comment:
These sample stats are massive, but it's the most convenient way to check expected metrics. It is much easier than adding `assert` for each path within a dictionary, and much easier to modify if we change the metrics generation.

Reply:
Agreed. You can move the test data into a separate file.